src/controllers/expenseController.js

```diff
@@ -0,0 +1,118 @@
+const expenseService = require('../services/expenseService');
+const userService = require('../services/userService');
+
+const getExpenses = (req, res) => {
+  try {
+    const { userId, categories, from, to } = req.query;
+
+    let expensesResult = expenseService.getAllExpense();
+
+    if (userId) {
+      expensesResult = expensesResult.filter(
+        (expense) => expense.userId === +userId,
+      );
+    }
+
+    if (categories) {
+      expensesResult = expensesResult.filter(
+        (expense) => expense.category === categories,
```

Comment:
The filtering logic for categories assumes `categories` is a single value. If `categories` can be a list (e.g., comma-separated), you should split the string and check if `expense.category` is included in the list.